packages/wallets/src/features/cfd/components/CFDPlatformsListAccounts/CFDPlatformsListAccounts.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useMemo } from 'react';
+import {
+    useActiveWalletAccount,
+    useAuthorize,
+    useCtraderAccountsList,
+    useDxtradeAccountsList,
+    useInvalidateQuery,
+    useSortedMT5Accounts,
+} from '@deriv/api-v2';
+import { TradingAppCardLoader } from '../../../../components/SkeletonLoader';
+import {
+    AddedCTraderAccountsList,
+    AddedDxtradeAccountsList,
+    AddedMT5AccountsList,
+    AvailableCTraderAccountsList,
+    AvailableDxtradeAccountsList,
+    AvailableMT5AccountsList,
+} from '../../flows';
+import { GetMoreMT5Accounts } from '../../screens';
+import './CFDPlatformsListAccounts.scss';
+
+type TProps = {
+    onMT5PlatformListLoaded?: (value: boolean) => void;
+};
+
+const CFDPlatformsListAccounts: React.FC<TProps> = ({ onMT5PlatformListLoaded }) => {
+    const { isFetching } = useAuthorize();
+    const {
+        areAllAccountsCreated,
+        data: mt5AccountsList,
+        isFetchedAfterMount: isMT5FetchedAfterMount,
+    } = useSortedMT5Accounts();
+    const { data: ctraderAccountsList } = useCtraderAccountsList();
+    const { data: dxtradeAccountsList, isFetchedAfterMount: isDxtradeFetchedAfterMount } = useDxtradeAccountsList();
+    const { data: activeWallet } = useActiveWalletAccount();
+    const invalidate = useInvalidateQuery();
+
+    const hasMT5Account = useMemo(() => {
+        return mt5AccountsList?.some(account => account.is_added);
+    }, [mt5AccountsList]);
+    const hasCTraderAccount = !!ctraderAccountsList?.length;
+    const hasDxtradeAccount = !!dxtradeAccountsList?.length;
+
+    useEffect(() => {
+        if (!isFetching) {
+            invalidate('mt5_login_list');
+            invalidate('trading_platform_accounts');
+        }
+    }, [invalidate, isFetching]);
+
+    useEffect(() => {
+        onMT5PlatformListLoaded?.(isMT5FetchedAfterMount);
+        return () => onMT5PlatformListLoaded?.(false);
+    }, [isMT5FetchedAfterMount, onMT5PlatformListLoaded]);
```

Comment:
I've seen 'fetchedAfterMount' usages previously and I'm pretty sure they will cause troubles the moment the data flow gonna change even at the slightest :)

Reply:
changed the implementation to use isLoading instead